src/cbc_sdk/platform/users.py

```diff
@@ -408,17 +420,22 @@ def _internal_add_profiles(self, profile_templates):
             profile_templates (list[dict]): List of profile templates to be added to the user.  Must be normalized.
         """
         grant = self.grant()
-        for template in profile_templates:
-            need_create = True
-            for profile in grant.profiles_:
-                if profile.matches_template(template) and profile.conditions['disabled']:
-                    profile.set_disabled(False)
-                    grant.touch()
-                    need_create = False
-                    break
-            if need_create:
-                grant.create_profile(template)
-        grant.save()
+        if grant:
+            for template in profile_templates:
+                need_create = True
+                for profile in grant.profiles_:
+                    if profile.matches_template(template) and profile.conditions['disabled']:
+                        profile.set_disabled(False)
+                        grant.touch()
+                        need_create = False
+                        break
+                if need_create:
+                    grant.create_profile(template)
+            grant.save()
+        else:
```

Comment:
This is the one place where I don't need to throw an error if there's no grant, because it's reasonable to create a new grant containing the profiles in question.